src/OpenTelemetry/Metrics/HistogramMeasurements.cs

```diff
@@ -41,5 +41,44 @@ internal HistogramMeasurements(double[] histogramBounds)
             this.AggregatedBucketCounts = histogramBounds != null ? new long[histogramBounds.Length + 1] : null;
             this.LockObject = new object();
         }
+
+        public Enumerator GetEnumerator() => new(this);
+
+        public struct Enumerator
+        {
+            private readonly bool isHistogramSumCount;
+            private readonly int numberOfBuckets;
+            private readonly int numberofExplicitBounds;
```

Comment:
If you look at the original one I did, I had a `Count` on the parent class which I used in `MoveNext`. The reason for that is it is better for perf to keep structs as small as possible. What I don't know is if that size-reduction provides more perf than caching these values in the struct to reference in each `MoveNext` invocation. Might be worth measuring and possibly adjusting (could be a follow-up).

At the least, I think we can remove `bool isHistogramSumCount` because `this.numberOfBuckets` will be 0 in that case so `MoveNext` will know to exit. Basically...

```csharp
            public bool MoveNext()
            {
                if (this.index < this.numberOfBuckets)
```

...should be enough (I think)?

Reply:
I have removed `idHistogramSumCount`. I will do the benchmarking for size-reduction vs caching later on.